src/wasm/wat-parser.cpp

```diff
@@ -350,9 +351,13 @@ struct ParseTypeDefsCtx {
   // Parse the names of types and fields as we go.
   std::vector<TypeNames> names;
 
-  // The index of the type definition we are parsing.
+  // The index of the type definition (i.e. rec group) we are parsing. This is
```

Comment:
Could this be rephrased as "index is the rec group index, typeIndex is the type index inside the rec group"? If so, index => recIndex maybe?

Reply:
I stopped using `parseDefs` for type parsing, which let me remove this useless field entirely and rename `typeIndex` to just `index`.